clang/lib/Sema/SemaDecl.cpp

```diff
@@ -14397,17 +14559,21 @@ void Sema::CheckCompleteVariableDeclaration(VarDecl *var) {
   QualType baseType = Context.getBaseElementType(type);
   bool HasConstInit = true;
 
+  if (getLangOpts().C23 && var->isConstexpr() && !Init)
+    Diag(var->getLocation(), diag::err_constexpr_var_requires_const_init)
+        << var;
```

Comment:
Is there a reason the existing logic below doesn't suffice and we need a separate code path here?

Reply:
Yes, existing logic uses `InitializationSequence` code paths that are deeply tied to C++ initialization rules, modifying these for C didn't seem reasonable.